test/unit/math/prim/mat/prob/VectorRNGTestRig.hpp

```diff
@@ -0,0 +1,255 @@
+                                             stan::math::negative_infinity(),
+                                             stan::math::not_a_number() };
+  /*
+   * This function wraps up the random number generator for testing.
+   *
+   * The tested rng can have up to three parameters. Any unused parameters can
+   * be ignored.
+   *
+   * It *must* be implemented in the child class (it isn't virtual here because
+   * C++ doesn't allow templated virtual member functions)
+   */
+  template<typename T1, typename T2, typename T3, typename T_rng>
+  auto generate_samples(const T1& p1, const T2& p2, const T3& p3,
+                        T_rng& rng) const;
+
+  /*
+   * This function builds the quantiles that we will supply to
+   * assert_matches_quantiles to test the random number generator implemented in
+   * generate_samples
+   */
+  virtual std::vector<double> generate_quantiles(double p1, double p2,
+                                                 double p3) const = 0;
+
+  VectorRNGTestRig(int N, int M,
+                   std::vector<double> good_p1,
+                   std::vector<int> good_p1_int,
+                   std::vector<double> bad_p1,
+                   std::vector<int> bad_p1_int,
+                   std::vector<double> good_p2,
+                   std::vector<int> good_p2_int,
+                   std::vector<double> bad_p2,
+                   std::vector<int> bad_p2_int,
+                   std::vector<double> good_p3,
+                   std::vector<int> good_p3_int,
+                   std::vector<double> bad_p3,
+                   std::vector<int> bad_p3_int) : N_(N), M_(M),
+                                                  good_p1_(good_p1),
+                                                  good_p1_int_(good_p1_int),
+                                                  bad_p1_(bad_p1),
+                                                  bad_p1_int_(bad_p1_int),
+                                                  good_p2_(good_p2),
+                                                  good_p2_int_(good_p2_int),
+                                                  bad_p2_(bad_p2),
+                                                  bad_p2_int_(bad_p2_int),
+                                                  good_p3_(good_p3),
+                                                  good_p3_int_(good_p3_int),
+                                                  bad_p3_(bad_p3),
+                                                  bad_p3_int_(bad_p3_int) {
+  }
+
+  VectorRNGTestRig(int N, int M,
+                   std::vector<double> good_p1,
+                   std::vector<int> good_p1_int,
+                   std::vector<double> bad_p1,
+                   std::vector<int> bad_p1_int,
+                   std::vector<double> good_p2,
+                   std::vector<int> good_p2_int,
+                   std::vector<double> bad_p2,
+                   std::vector<int> bad_p2_int) : N_(N), M_(M),
```

Comment:
[optional]

Put the `:` on a new line.  You also don't want to waste horizontal space like this.